components/motor/gpio/basic.go

```diff
@@ -106,8 +107,9 @@ type Motor struct {
 	dirFlip                  bool
 	motorName                string
 
-	opMgr  operation.SingleOperationManager
-	logger golog.Logger
+	opMgr   operation.SingleOperationManager
+	logger  golog.Logger
+	stateMu sync.RWMutex
```

Comment:
[minor] This is unused now, the only state we keep track of in the motor is `m.on`, it has been operating fine without it, and if we decide to add mutex to the basic motor I can add read and write locks similar to the encoded motor. This will mean additional testing for this PR.

Reply:
Do `on` and `powerPct` need to be changed together? If so, I think we might need to use a mutex. My gut instinct is to have a garden-variety mutex instead of an RW one, but I can't justify that other than with "we're unlikely to have multiple concurrent long-running reads, so we won't use the RWMutex's advantages but will incur their disadvantages any time it gets locked for reading _or_ writing."

...stepping back a moment, could `m.on` be replaced with `m.powerPct != 0`, and then we could get rid of some mutable state? 

In case you don't see the concern re: `on` and `powerPct`: suppose we call `setPWM` simultaneously in two goroutines, one for 0% and one for 100%. It's possible to have this order of events: the first call sets powerPct to 0, then the second call sets it to 100, then the second call sets `on` to true, and finally the first call sets it to false. The end result is that we're at 100% power and `on` is false. If we grabbed a mutex at the beginning, we could never have these overlapping side effects, and the state would remain consistent.

Reply:
I see the concern, but unfortunately `powerPct` is a float and can be negative, so we actually have a lot of checks on `math.Abs(m.powerPct) <=/>= bound`all over the motor code. t is necessary for motor powers to be floats.

 I feel like it might be a bit of a refactor that's getting into the redesign of motor/motor_encoder. My thoughts were that we're already checking for the `powerPct` in that condition before setting the motor state to false, and then going through the `Stop` logic with PWM pins.
 
 Happy to have another suggestion and go through that if you see a workable one.

Reply:
@steve, I started this thread to ask if I should lock it, Alan had no strong feelings, so I was going to remove them. There were no mutexes on `basic.go`